__tests__/AccountSettings.js

```diff
@@ -0,0 +1,77 @@
+import {
+  AWSInfoResponse,
+  getPersistedMockCall,
+  userResponse,
+} from 'testUtils/mockHttpCalls';
+import { renderRouteWithStore } from 'testUtils/renderUtils';
+
+const elementIDs = {
+  ChangeEmailForm: 'account-settings/change-email',
+  ChangePasswordForm: 'account-settings/change-password',
+  DeleteAccount: 'account-settings/delete-account',
+
+  EmailLabel: 'account-settings/email-label',
+  EmailDescription: 'account-settings/email-description',
+};
+
+const requests = {};
+
+describe('AccountSettings', () => {
+  beforeAll(() => {
+    requests.user = getPersistedMockCall('/v4/user/', userResponse);
+    requests.info = getPersistedMockCall('/v4/info/', AWSInfoResponse);
+    requests.orgs = getPersistedMockCall('/v4/organizations/');
+    requests.clusters = getPersistedMockCall('/v4/clusters/');
+    requests.catalogs = getPersistedMockCall('/v4/appcatalogs/');
+  });
+
+  afterAll(() => {
+    for (const req of Object.values(requests)) {
+      req.persist(false);
+    }
+  });
+
+  it('renders the account settings component on the "/account-settings" route', async () => {
+    const { findByText } = renderRouteWithStore('/account-settings', {});
+
+    await findByText('Your Account Settings');
+  });
+
+  it('renders the email change form', async () => {
+    const { findByTestId } = renderRouteWithStore('/account-settings', {});
+
+    const emailChangeForm = await findByTestId(elementIDs.ChangeEmailForm);
+
+    expect(emailChangeForm).not.toBeNull();
+  });
+
+  it('renders the email change form label and description', async () => {
+    const { findByTestId } = renderRouteWithStore('/account-settings', {});
+
+    const emailLabel = await findByTestId(elementIDs.ChangeEmailForm);
+    expect(emailLabel).not.toBeNull();
+
+    const emailDescription = await findByTestId(elementIDs.ChangeEmailForm);
+    expect(emailDescription).not.toBeNull();
+  });
+
+  it('renders the password change form', async () => {
+    const { findByTestId } = renderRouteWithStore('/account-settings', {});
+
+    const passwordChangeForm = await findByTestId(
+      elementIDs.ChangePasswordForm
+    );
+
+    expect(passwordChangeForm).not.toBeNull();
```

Comment:
Shouldn't we be looking for the right value to be rendered instead?

Reply:
I'm not sure what you mean by that
Actual tests for the `PasswordChangeForm` are in a separate file, this just checks that it's there

Reply:
I find it confusing and not adding a lot of value to a test, especially the `not.toBeNull()` part.

To be clear, my two points are: I think there are too many`not.toBeNull`s and `findById`s in these tests.

Reply:
I agree that `not.toBeNull()` is not adding much value, other than readibility. But the `findById` calls are checking if the elements are rendered, so they definately have a value.